### PR TITLE
improvement documentation for record attribution

### PR DESCRIPTION
**Pull request by manikandan-ravikumar**
## Summary
This PR improves the existing documentation for `record attribution` [here](https://docs.nosto.com/techdocs/apis/js-apis/common-examples/record-attributions). This PR adds the following details
1. List of all event types that's supported
2. Adds more details on parameters of every example

## JIRA
[ADS-3225](https://nostosolutions.atlassian.net/browse/ADS-3255)

### CODE REVIEW

File: apis/js-apis/common-examples/record-attributions.md [hunk 1]
```diff
@@ -31,8 +49,13 @@ nostojs(function(api) {
   .done()
 });
 ```
+In the above example,
+- `vp` specifies the type of event and it corresponds to View Product
+- `12345678` specifies the target and it corresponds to the ID of the product that's being viewed
+- `frontpage-nosto-1` specifies tge reference and it corresponds to the placement that hosted the product that's being clicked
```
Comment (gary-nosto) [inline apis/js-apis/common-examples/record-attributions.md:53]:
```suggestion
- `frontpage-nosto-1` specifies the reference and it corresponds to the placement that hosted the product that's being viewed
```

typo

File: apis/js-apis/common-examples/record-attributions.md [hunk 2]
```diff
@@ -31,8 +49,13 @@ nostojs(function(api) {
...
 
-(2) Attributing a placement click to a `cc` event
+
+2. Attributing a placement click to a `cc`  (Custom Campaign) event
```
Comment (gary-nosto) [inline apis/js-apis/common-examples/record-attributions.md:56]: Can you explain what a custom campaign is? I'm not sure what it is either 😅

File: apis/js-apis/common-examples/record-attributions.md [hunk 3]
```diff
@@ -42,8 +65,12 @@ nostojs(function(api) {
   .done()
 });
 ```
+In the above example,
+- `cc` specifies the type of event and it corresponds to Custom Campaign
+- `12345678` specifies the target and it corresponds to the ID of the product that's being viewed
+- `frontpage-nosto-1` specifies tge reference and it corresponds to the placement that hosted the product that's being clicked
```
Comment (gary-nosto) [inline apis/js-apis/common-examples/record-attributions.md]:
```suggestion
- `frontpage-nosto-1` specifies the reference and it corresponds to the placement that hosted the product that's being viewed
```

typo

File: apis/js-apis/common-examples/record-attributions.md [hunk 4]
```diff
@@ -52,6 +79,12 @@ nostojs(api => api
 .done()
 ```
 
-In the above example, we are recording a `View Product` event for product 7513863258337 which was clicked from the recommendation slot `productpage-nosto-3` while on another product page `7513872007393`.&#x20;
+In the above example,
+- `vp` specifies the type of event and it corresponds to View Product
+- `7513863258337` specifies the target and it corresponds to the ID of the product that's being viewed
+- `productpage-nosto-3` specifies the reference and it corresponds to the placement that hosted the product that's being clicked
+- `7513872007393` specifies the reference source and it corresponds to the ID of element, of the current page, that contained the `ref` element, `productpage-nosto-3` in this case
```
Comment (gary-nosto) [inline apis/js-apis/common-examples/record-attributions.md]:
```suggestion
- `7513872007393` specifies the reference source and it corresponds to the ID of element, of the current page, that contained the `ref` element, `productpage-nosto-3` in this case
```

`refSrc` for `vp` events refers to the id of the product that was shown where the recommendation was shown.  I think your original explanation explained it better.

File: apis/js-apis/common-examples/record-attributions.md (hunk 2 above)
Comment (manikandan-ravikumar) [reply apis/js-apis/common-examples/record-attributions.md:56]: Done

File: apis/js-apis/common-examples/record-attributions.md (hunk 3 above)
Comment (manikandan-ravikumar) [reply apis/js-apis/common-examples/record-attributions.md]: Done

File: apis/js-apis/common-examples/record-attributions.md (hunk 4 above)
Comment (manikandan-ravikumar) [reply apis/js-apis/common-examples/record-attributions.md]: improved the description

**Review by gary-nosto — APPROVED** [review]
Looks good!

**Pull request merged**